rb/lib/selenium/webdriver/common/logger.rb

```diff
@@ -122,7 +122,7 @@ def create_logger(output)
         logger.progname = 'Selenium'
         logger.level = default_level
         logger.formatter = proc do |severity, time, progname, msg|
-          "#{time.strftime('%F %T')} #{severity} #{progname} #{msg}\n"
+          "#{time.strftime('%F %T')} #{severity} #{progname} #{msg.try(:force_encoding, 'UTF-8')}\n"
```

Comment:
I've been working in this area (Encoding), a lot recently. Can you try the following code.

```
def utf8?
  Encoding.default_internal == Encoding::UTF_8 &&
    Encoding.default_external == Encoding::UTF_8
end

# Then in your block
msg = msg.force_encoding('UTF-8') if msg.respond_to?(:force_encoding)
```

Now to the caveats. ASCII-8BIT is a useful encoding char-set because it allows things like german umlauts and other stuff. I actually have the following code in my new API tests I write

```
# file - env.rb
# Set these to 8bit to allow umlauts
Encoding.default_external = Encoding::ASCII_8BIT
Encoding.default_internal = Encoding::ASCII_8BIT
```

What is your current `Encoding.default_external / internal` values, could it be your system is somehow setting up some weird ones?